mirrorlist/mirrorlist_client.wsgi

```diff
@@ -165,7 +165,7 @@ def application(environ, start_response):
         text = ""
         text += message + '\n'
         for (hostid, url) in results:
-            text += url + '\n'
+            text += url[0] + '\n'
```

Comment:
This is described in the first commit. To trim the protocols to only contain the requested protocols I am re-using the function trim_to_preferred_protocols(). This function is currently called after the metalink is created and changes the result from [(hostid, [url, url, url]), ...] to [(hostid, url), ...], which was okay for the mirrorlist back-end as it only returned one URL per host. As the function trim_to_preferred_protocols() already reduces the URLs returned per host to a defined set (https, http, ftp) for metalink I wanted to re-use this function to trim the URL list to the requested protocol.

To be able to re-use trim_to_preferred_protocols() it needs to return a result which metalink and mirrorlist can handle. metalink wants multiple URLs per host and therefore I am now always returning [(hostid, [url, url, url]), ...] and [url, url, url] now always contains exactly one element for mirrorlist and therefore url[0].